discovery-infra/tests/base_test.py

```diff
@@ -90,12 +87,45 @@ def assert_string_length(string, expected_len):
         assert len(string) == expected_len, "Expected len string of: " + str(expected_len) + \
                                             " rather than: " + str(len(string)) + " String value: " + string
 
-    @staticmethod
-    def collect_test_logs(cluster, api_client, test):
+    def collect_test_logs(self, cluster, api_client, test, nodes):
         log_dir_name = f"{env_variables['log_folder']}/{test.name}"
         with suppress(ApiException):
             cluster_details = json.loads(json.dumps(cluster.get_details().to_dict(), sort_keys=True, default=str))
             download_logs(api_client, cluster_details, log_dir_name, test.result_call.failed)
+        if test.result_call.failed:
+            self._collect_virsh_logs(nodes, log_dir_name)
+
+    @staticmethod
+    def _collect_virsh_logs(nodes, log_dir_name):
+        logging.info('---TEARDOWN --- Collecting virsh logs\n')
+        if not os.path.exists(log_dir_name):
+            logging.info(f"Creating virsh logs directory: {log_dir_name}")
+            os.makedirs(log_dir_name, exist_ok=False)
```

Comment:
You can keep switch to `exist_ok=True` and remove the previous `os.path.exists` check

Reply:
exists_ok True\ false is kind of irrelevant, no? cause im already checking if this folder doesnt exist, and than run create dir

Reply:
That's right. But to make the code cleaner you can just keep it simple without any conditions.
```python
def _collect_virsh_logs(nodes, log_dir_name):
    logging.info('Collecting virsh logs\n')
    os.makedirs(log_dir_name, exist_ok=True)
    ...
```